providers/src/airflow/providers/apache/spark/operators/spark_sql.py

```diff
@@ -94,15 +91,6 @@ def __init__(
         self._yarn_queue = yarn_queue
         self._hook: SparkSqlHook | None = None
 
-    @property
-    @deprecated(
-        reason="`_sql` is deprecated and will be removed in the future. Please use `sql` instead.",
-        category=AirflowProviderDeprecationWarning,
-    )
-    def _sql(self):
-        """Alias for ``sql``, used for compatibility (deprecated)."""
-        return self.sql
```

Comment:
That is a bit odd?
private functions don't require deprecation. I assume this was a workaround to deprecate `sql` not `_sql`
can you investigate? (git blame to find the previous PRs might help to see what was the intention)

Reply:
So @eladkal, it seems `_sql()` was marked for deprecation and `sql` was introduced to comply with the templated field validations - https://github.com/apache/airflow/pull/38045

Reply:
Ah OK.. a piece of history :D
Lets add some clarification in the change log message that `_sql` was introduced in 2016 and since it was listed in as templated field we handled it as public api despite the `_` prefix that marked it as private.

(This is just so it wouldn't look odd why we create a breaking change version to remove private parameter)

Reply:
Added this!